upgrade_test.py

```diff
@@ -31,50 +31,105 @@ class UpgradeTest(FillDatabaseData):
 
     :avocado: enable
     """
+    orig_ver = None
+    new_ver = None
 
     def upgrade_node(self, node):
         repo_file = self.db_cluster.params.get('repo_file', None,  'scylla.repo.upgrade')
         new_version = self.db_cluster.params.get('new_version', None,  '')
         upgrade_node_packages = self.db_cluster.params.get('upgrade_node_packages')
+        upgrade_mode = self.db_cluster.params.get('upgrade_mode', default='smooth')
```

Comment:
mode shouldn't be a param, it should be defined according to the base version and the target target version (can parse it from the repo target).
Something in the spirit of:
if base == opensource & target == enterprise:
   mode = reinstall (direct)
else:
   mode = smooth